agent/proxycfg/testing_tproxy.go

```diff
@@ -655,3 +655,117 @@ func TestConfigSnapshotTransparentProxyDestination(t testing.T) *ConfigSnapshot
 		},
 	})
 }
+
+func TestConfigSnapshotTransparentProxyDestinationHTTP(t testing.T) *ConfigSnapshot {
+	// DiscoveryChain without an UpstreamConfig should yield a
+	// filter chain when in transparent proxy mode
+	var (
+		google    = structs.NewServiceName("google", nil)
+		googleUID = NewUpstreamIDFromServiceName(google)
+		googleCE  = structs.ServiceConfigEntry{Name: "google", Destination: &structs.DestinationConfig{Addresses: []string{"www.google.com"}, Port: 443}, Protocol: "http"}
+
+		kafka    = structs.NewServiceName("kafka", nil)
+		kafkaUID = NewUpstreamIDFromServiceName(kafka)
+		kafkaCE  = structs.ServiceConfigEntry{Name: "kafka", Destination: &structs.DestinationConfig{Addresses: []string{"192.168.2.1"}, Port: 9093}, Protocol: "http"}
+	)
+
+	return TestConfigSnapshot(t, func(ns *structs.NodeService) {
```

Comment:
`TestConfigSnapshot` would really benefit from a `populateServices` bool like `TestConfigSnapshotTerminatingGateway` and `TestConfigSnapshotIngressGateway` have to avoid adding base services that don't have anything to do with the test (just add clutter to the golden files generated). This is just a thought though, definitely not a blocker to merging.

Reply:
I will make sure to add that in the test clean up PR after the release.